src/websocket/admin.ts

```diff
@@ -34,7 +34,7 @@ export class AdminWebsocket implements Api.AdminApi {
 
   _requester = <ReqO, ReqI, ResI, ResO>(tag: string, transformer?: Transformer<ReqO, ReqI, ResI, ResO>) =>
     requesterTransformer(
-      req => this.client.request(req).then(catchError),
+      (req, timeout) => promiseTimeout(this.client.request(req), timeout).then(catchError),
```

Comment:
```suggestion
      (req, timeout) => promiseTimeout(this.client.request(req).then(catchError), timeout),
```

Reply:
These changes will have some Typescript errors. 
Also If the `Promise.race` throws an error it will wait for the other condition it complete (i.e. the timeout) which we do not want.

Also, I don't see the catch scope as very limited because it also handles errors when we are not connected to the Websocket. So I think it is able to handle the timeout error should not be an issue.
What do you think?

Reply:
It's OK if the timeout `reject` gets called after the main promise is resolved, it just won't do anything after the overall race promise is resolved via the main promise

Reply:
I will trust Michaels review of it, he's the original author